src/main/java/com/poli/internship/api/controller/ApplicationController.java

```diff
@@ -32,19 +33,30 @@ public Application getApplicationById(@Argument Map input, GraphQLContext ctx) {
         GraphQLAuthorization.checkAuthorization(ctx);
 
         Map data = (Map) input.get("input");
-        return this.getApplicationByIdUseCase.exec((String) data.get("id"));
+        Application application = this.getApplicationByIdUseCase.exec((String) data.get("id"));
+        if(UserType.valueOf(ctx.get("userType")) == UserType.STUDENT && !((String)ctx.get("userId")).equals(application.userId())) {
+            throw new CustomError("Not current student application.", ErrorType.UNAUTHORIZED);
+        }
+
+        return application;
```

Comment:
- eu colocaria a checagem pra ver se application é do student mesmo no use case em vez daqui, mas tanto faz
- acho que o tipo de erro certo é FORBIDDEN, em vez de UNAUTHORIZED

"401 Unauthorized is the status code to return when the client provides no credentials or invalid credentials. 403 Forbidden is the status code to return when a client has valid credentials but not enough privileges to perform an action on a resource."

https://auth0.com/blog/forbidden-unauthorized-http-status-codes/

Reply:
tbm teria que alterar o tipo de erro na chamada de delete

Reply:
1) Estou deturpando o padrão de projeto?